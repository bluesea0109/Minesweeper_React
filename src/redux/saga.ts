import {takeLatest} from 'redux-saga/effects';
import {
  watchChannel,
  ReceivingData,
} from '../feature/featureAction';
import {
  newRound,
  initRound,
} from '../feature/featureReducer';

export function* watcherSaga() {
  yield takeLatest(initRound.type, watchChannel);
  yield takeLatest(newRound.type, ReceivingData);
}
