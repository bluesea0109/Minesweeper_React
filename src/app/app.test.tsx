import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import App from "./app";
import { mount } from "enzyme";
import { cleanup, render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import createSagaMiddleware from "redux-saga";
import { watcherSaga } from "../redux/saga";

const sagaMiddleware = createSagaMiddleware();
const mockStore = configureStore([sagaMiddleware]);

describe("<App />", () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });
  it("Rendering test!", () => {
    const initialState = {
      feature: {
        map: [],
        mineSign: [],
        message: "",
      },
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Selecting difficulty", () => {
    const initialState = {
      feature: {
        map: [],
        mineSign: [],
        message: "",
      },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(wrapper.text().includes("Minesweeper")).toBe(true);
    expect(wrapper.text().includes("10 ✕ 10")).toBe(true);
    expect(wrapper.text().includes("40 ✕ 20")).toBe(true);
    expect(wrapper.text().includes("100 ✕ 50")).toBe(true);
  });
});
