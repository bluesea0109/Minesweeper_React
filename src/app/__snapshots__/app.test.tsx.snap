// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Rendering test! 1`] = `
<div
  className="makeStyles-layout-3"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 makeStyles-container-4 css-p7cgzs-MuiPaper-root"
  >
    <div
      className="makeStyles-headText-1"
    >
      Minesweeper
    </div>
    <div
      className="makeStyles-boxContainer-11"
    >
      <div
        className="makeStyles-boxes-10 MuiBox-root css-0"
        onClick={[Function]}
      >
        10 ✕ 10
      </div>
      <div
        className="makeStyles-boxes-10 MuiBox-root css-0"
        onClick={[Function]}
      >
        40 ✕ 20
      </div>
      <div
        className="makeStyles-boxes-10 MuiBox-root css-0"
        onClick={[Function]}
      >
        100 ✕ 50
      </div>
      <div
        className="makeStyles-boxes-10 MuiBox-root css-0"
        onClick={[Function]}
      >
        500 ✕ 50
      </div>
    </div>
  </div>
</div>
`;
